veejay-current/veejay-server/veejay/vj-perform.c

```diff
@@ -1127,32 +1127,33 @@ void vj_perform_free(veejay_t * info)
 
     sample_record_free();
 
-    if(info->edit_list->has_audio)
+    if(info->edit_list && info->edit_list->has_audio)
         vj_perform_close_audio();
 
-    if( fx_chain_buffer == NULL ) {
+    if( fx_chain_buffer != NULL ) {
```

Comment:
fx_chain_buffer is allocated at the start of veejay, when it is NULL, veejay will dynamically allocate buffers for FX rendering.  (see vj_perform_alloc_row, vj_perform_free_row), why this change?

Reply:
* Firstly, yes this fix a segfault (SIGSEGV Invalid memory reference)
There is no memory access control in `vj_perform_row_used` macro :
`vj-perform.c:713: #define vj_perform_row_used(c) ( frame_buffer[c]->Y == NULL ? 0 : 1 )`
hotow reproduce : `veejay -g file-that-doesnt-exist.ext`

**I did a mistake, the memory check must be in done the macro :** 
`#define vj_perform_row_used(c) ( frame_buffer == NULL ? 0 : (frame_buffer[c]->Y == NULL ? 0 : 1 ) )`

* Second (found that reviewing) - logical error ? 
`if fxchainbuff == NULL do (if fxchainbuff != NULL)`
vj-perform.c:1133
```
    if( fx_chain_buffer == NULL ) { <------ if chain_buffer == NULL -> do something
       for (c = 0; c < fblen; c++) {
        if(vj_perform_row_used(c))
            vj_perform_free_row(c);

```
[...]
vj-perform.c:692
```
static void vj_perform_free_row(int c)
{
    if( fx_chain_buffer ) <------ if chain_buffer == NULL -> return
        return;
```

Reply:
````if( fx_chain_buffer == NULL ) { <------ if chain_buffer == NULL -> do something````

Yes, it should only enter the loop if fx_chain_buffer was not allocated. Veejay has an option to dynamically allocate memory needed for the FX chain, by default, it allocates enough memory at startup to keep the whole FX chain preallocated in memory (see info->uc->ram_chain).
Thus, when the fx_chain_buffer is NULL, veejay should dynamically allocate memory for the FX entry

Reply:
So, fx_chain_buffer == NULL, is the correct version ;)

Reply:
```#define vj_perform_row_used(c)```

There is no need to test for frame_buffer[c], this is allocated in vj_perform_init and mlock'ed; 
In case of dynamic effect chain, frame_buffer[c]->Y is allocated and free'd when needed.

Reply:
I'll merge this in, and then push some fixes on this pull request!